src/components/lib/cos_kernel_api.c

```diff
@@ -640,6 +644,79 @@ cos_sinv(sinvcap_t sinv, word_t arg1, word_t arg2, word_t arg3, word_t arg4)
 	return call_cap_op(sinv, 0, arg1, arg2, arg3, arg4);
 }
 
+/*
+ * TODO:
+ * talk about this naming choice
+ * can't remember what Gabe thought was best choice here
+ */
+void *
+cos_hypervisor_get_resource(int op, void *arg1, void *arg2, void *arg3)
+{
+	void *ret;
+	assert(op >= 0);
+
+	/* Dpending on the operation we will need to allocate our own a cap space */
```

Comment:
Don't you have spell check on in your editor for comments?  Or does `vim` not have that...

Reply:
kek